### ISSUE TITLE
Attempting to use project with latest v2 endpoints

### ISSUE THREAD

**Issue by gep13** [jamiemaguiredotnet/SocialOpinion-Public#26]
**Describe the bug**

I am trying to use this library to post a tweet via the new v2 Twitter endpoint, and I either get a 403 or 401 response.  What I can't determine if I am holding this wrong, or there is something I am missing.

I have added what I "think" is the right information from the Twitter Developer Portal, namely:

<img width="487" alt="image" src="https://github.com/jamiemaguiredotnet/SocialOpinion-Public/assets/1271146/ce94916e-bc51-4e23-940e-7618f1183704">

but I am not sure if they are the right ones.

Do you have a complete set of steps for using this library with the new Twitter v2 endpoint?

**To Reproduce**
1. Create app.config file, and copy/paste values from above section of Twitter Developer Portal
2. Comment out the sections of the program.cs leaving only the lines for posting a new Tweet
3. Run the application
4. Get either a 403 or 401 response, depending on what keys I use from the Twitter Developer Portal

**Expected behavior**

I would love to be able to correctly post a new Tweet to Twitter :smile:

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Mac
 - Browser: Chrome
 - Version: Latest

**Additional context**

**Comment by gep13** [comment]
Digging into the code a little but here, I think I am right in saying that the TweetService.PostTweetV2` method is calling the `TweetClient.PostTweet` method, which is using the "old" URL of https://api.twitter.com/1.1/statuses/update.json, when it should be calling into https://api.twitter.com/2/tweets.

Does that make sense?  Or am I missing something here?

Thanks

**Comment by a-gubskiy** [comment]
@gep13 thank you for so detail explanation! I prepared PR with fix for this issue. I hope @jamiemaguiredotnet will approve and merge it and we will have new version of this useful library soon!

**Comment by jamiemaguiredotnet** [comment]
> Digging into the code a little but here, I think I am right in saying that the TweetService.PostTweetV2`method is calling the`TweetClient.PostTweet` method, which is using the "old" URL of https://api.twitter.com/1.1/statuses/update.json, when it should be calling into https://api.twitter.com/2/tweets.
> 
> Does that make sense? Or am I missing something here?
> 
> Thanks

Thankyou for the feedback.

Yes. The older endpoint was being used at it supported upload of Media / Images whereas V2 did not (at the time of initial implementation).

It should be ok to use V2 now. I can see @gep13 has prepared a PR.  Thankyou! :)

**Comment by gep13** [comment]
> @jamiemaguiredotnet said...
I can see @gep13 has prepared a PR. Thankyou! :)

I can't take credit for the PR, this was done by @ernado-x 😄  Thank you for getting this merged!

**Comment by jamiemaguiredotnet** [comment]
Whilst I remember.

I noticed the permission issue still occurs.

It looks like it is because of 2 things:

 - OAuth signature in POST requests with JSON  body.
 - Change at Twitter API end handling POST


I have found a workaround and been able to post successfully using the V2 endpoint but it meant bringing in a couple of new NuGet packages.

If I get time at the weekend I will try to merge these fixes in.
